src/LoginContext.js

```diff
@@ -0,0 +1,18 @@
+import React, { createContext } from "react";
+
+export const LoginStatusContext = createContext();
+export const LoginClickContext = createContext();
+
+export function LoginProviders({ children, isLoggedIn, setIsLoggedIn }) {
+  const handleLoginClick = () => {
+    setIsLoggedIn((prevState) => !prevState);
+  };
+
+  return (
+    <LoginStatusContext.Provider value={isLoggedIn}>
+      <LoginClickContext.Provider value={handleLoginClick}>
+        {children}
+      </LoginClickContext.Provider>
+    </LoginStatusContext.Provider>
```

Comment:
Context を分けていますが、一つにまとめた方が良いかと思いました！

以下のようなイメージです〜！

```suggestion
   const context = {
     isLoggedIn,
     handleLoginClick
   }

    <AuthContext.Provider value={context}>
        {children}
    </AuthContext.Provider>
```

Reply:
ご指摘いただいたように`AuthContext`として１つにまとめました！

Reply:
good!